Data.Access.Layer/Repositories/GenericRepository.cs

```diff
@@ -1,54 +1,66 @@
 ﻿using Data.Access.Layer.Data;
+using Data.Access.Layer.Models.Contracts;
 using Data.Access.Layer.Repositories.Contracts;
 using Microsoft.EntityFrameworkCore;
 using System.Linq.Expressions;
 
 
 namespace Data.Access.Layer.Repositories
 {
-    public class GenericRepository<T> : IGenericRepository<T> where T : class, new()
+    public class GenericRepository<TEntity,TDbContext> : IGenericRepository<TEntity> where TEntity : class, IUser where TDbContext : DbContext
     {
 
-        private readonly SqlDataContext _context;
+        private readonly TDbContext _context;
```

Comment:
When naming variables be free to have them in full name "TableDatabaseContext" instead of "TDbContext" as that would be much clearer and easier to read.
This is just a suggestion where it would be just fine to leave it as it is

Reply:
Done